pkg/poc/miseAuthentication.go

```diff
@@ -0,0 +1,74 @@
+package poc
+
+import (
+	"bytes"
+	"context"
+	"fmt"
+	"log"
+	"net/http"
+	"strings"
+)
+
+type MiseRequestData struct {
+	MiseURL        string
+	OriginalURI    string
+	OriginalMethod string
+	Token          string
+}
+
+const (
+	miseURL   = "http://localhost:5000/ValidateRequest"
+	originURI = "https://server/endpoint"
+)
+
+func authenticateWithMISE(w http.ResponseWriter, r *http.Request) error {
```

Comment:
To make `authenticateWithMISE` simpler, let's do this:

1. Change the function arguments to:

`func authenticateWithMISE(ctx context.Context, token string) error

- You can get context from the `http.Request` in `frontend.go` line 56
- We can use the token extract function in `frontend.go`

2. Based on the MISE response, we return an error value. No need to call `w.WriteHeader(resp.StatusCode)`, we'll leave that to frontend.go http handler.